sdk/dapp-kit/src/hooks/wallet/useDisconnectWallet.ts

```diff
@@ -0,0 +1,50 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+import type { UseMutationOptions } from '@tanstack/react-query';
+import { useMutation } from '@tanstack/react-query';
+import { useWalletContext } from '../../components/WalletProvider.js';
+import { walletMutationKeys } from '../../constants/walletMutationKeys.js';
+import { WalletNotConnectedError } from '../../errors/walletErrors.js';
+
+type UseDisconnectWalletMutationOptions = Omit<
+	UseMutationOptions<void, Error, void, unknown>,
+	'mutationFn'
+>;
+
+/**
+ * Mutation hook for disconnecting from an active wallet connection, if currently connected.
+ */
+export function useDisconnectWallet({
+	mutationKey,
```

Comment:
We should make this optional and have a sensible default

Reply:
Yep so the mutation key is optional, we allow folks to supply their own additional mutation keys with this setup.

```
useDisconnectWallet() => mutationKey = [{ baseScope: 'wallet', baseEntity: 'disconnect-wallet' }]
```

```
useDisconnectWallet({mutationKey: ['my-additional-key'] }) => mutationKey = [{ baseScope: 'wallet', baseEntity: 'disconnect-wallet' }, 'my-additional-key']
```